modules/javafx.controls/src/main/java/javafx/scene/control/DatePicker.java

```diff
@@ -147,11 +148,23 @@ public DatePicker(LocalDate localDate) {
 
         focusedProperty().addListener(o -> {
             if (!isFocused()) {
-                commitValue();
+                commitValueOnFocusLost();
             }
         });
     }
 
+    /**
+     * This is to address https://bugs.openjdk.org/browse/JDK-8303478
+     */
+    protected void commitValueOnFocusLost(){
+        try {
+            commitValue();
+        } catch (DateTimeParseException dtpe) {
+            this.requestFocus();
```

Comment:
I'm not sure whether moving focus to the control is a good idea, because this effectively captures focus and forces me to enter a valid date before I can do anything else. Why not just restore the previous value instead?

Reply:
I have changed the behavior as you suggested.
The downside is that user might not notice that his/her changes were replaced, but it is anyway consistent to what happens when typing ENTER.

Robot test is updated accordingly.